aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp

```diff
@@ -2716,6 +2716,54 @@ void dequantize_tensor_per_channel_float_qparams_cpu(
       });
 }
 
+void quantize_tensor_per_tensor_affine_sub_byte_cpu(
+    Tensor rtensor,
+    Tensor qtensor,
+    float scale,
+    float zero_point) {
+  check_tensor_memory_format(rtensor, qtensor);
+  const float* const rdata = rtensor.data_ptr<float>();
+  uint8_t* qdata = reinterpret_cast<uint8_t*>(qtensor.data_ptr<at::quint4x2>());
+  // TODO: get bitrate from dispatch based on dtype
+  auto numel = rtensor.numel();
+  auto BIT_RATE = 4;
+  const auto ELEM_PER_BYTE = 8 / BIT_RATE;
+
+  for (int i = 0; i < numel; ++i) {
+    float inv_scale = scale == 0 ? 1.0f : 1.0f / scale;
+    int64_t qvalue = lrintf(std::nearbyint(rdata[i] * inv_scale) + zero_point);
+    qvalue = std::max(0, std::min(qvalue, (1 << BIT_RATE) - 1));
```

Comment:
Is it possible to generalize this to support for un-signed and signed 4 bit dtypes?

Reply:
Yes, should be possible. Can add qmin and qmax in the dispatch args.